README.md

```diff
@@ -103,6 +103,26 @@ See the section: [How does it work? - Protocol Description](#how-does-it-work-pr
 <cidv1> ::= <multibase-prefix><cid-version><multicodec-packed-content-type><multihash-content-address>
 ```
 
+## Decoding Algorithm
+
+To decode a CID, follow the following algorithm:
+
+1. If it's a string (ASCII/UTF-8), decode it into raw bytes:
+  * If it is 46 characters long starts with `Qm...`, it's a CIDv0 without a multibase prefix. Decode it as base58btc.
+  * Otherwise, decode it according to the multibase spec.
+2. Given the decoded (binary) CID (`cid`):
+   * If it's 34 bytes long with the leading bytes `[0x12, 0x20, ...]`, it's a CIDv0.
+     * The CID's multihash is `cid`.
+     * The CID's multicodec is DagProtobuf
+     * The CID's version is 0.
+   * Otherwise, let `N` be the first varint in `cid`. This is the CID's version.
+     * If `N == 1` (CIDv1):
+       * THe CID's multicodec is the second varint in `cid`
```

Comment:
`s/THe/The`

Reply:
Here is the algo JS is doing: https://github.com/ipld/js-cid/blob/master/src/index.js#L18-L87 Some different heuristics, it trusts less the length and relies more on value checking.